# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/validationESEXT_autogen.h
# Commit: 13282d7a47a5
# Full Hash: 13282d7a47a57a73be8d060efd9645b4d642f03e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-09-08 09:46:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 3064469c073d -r 13282d7a47a5 gfx/angle/checkout/src/libANGLE/validationESEXT_autogen.h
--- a/gfx/angle/checkout/src/libANGLE/validationESEXT_autogen.h	Sat Sep 07 13:36:46 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/validationESEXT_autogen.h	Sat Sep 07 13:38:36 2019 +0000
@@ -110,7 +110,7 @@
                                              GLsizei drawcount);
 
 // GL_ANGLE_provoking_vertex
-bool ValidateProvokingVertexANGLE(Context *context, ProvokingVertex modePacked);
+bool ValidateProvokingVertexANGLE(Context *context, ProvokingVertexConvention modePacked);
 
 // GL_ANGLE_request_extension
 bool ValidateRequestExtensionANGLE(Context *context, const GLchar *name);
@@ -249,7 +249,7 @@
                                       GLsizei bufSize,
                                       const void *pixels);
 bool ValidateTexImage3DRobustANGLE(Context *context,
-                                   TextureType targetPacked,
+                                   TextureTarget targetPacked,
                                    GLint level,
                                    GLint internalformat,
                                    GLsizei width,
@@ -261,7 +261,7 @@
                                    GLsizei bufSize,
                                    const void *pixels);
 bool ValidateTexSubImage3DRobustANGLE(Context *context,
-                                      TextureType targetPacked,
+                                      TextureTarget targetPacked,
                                       GLint level,
                                       GLint xoffset,
                                       GLint yoffset,
@@ -295,7 +295,7 @@
                                                 GLsizei dataSize,
                                                 const GLvoid *data);
 bool ValidateCompressedTexImage3DRobustANGLE(Context *context,
-                                             TextureType targetPacked,
+                                             TextureTarget targetPacked,
                                              GLint level,
                                              GLenum internalformat,
                                              GLsizei width,
@@ -306,7 +306,7 @@
                                              GLsizei dataSize,
                                              const GLvoid *data);
 bool ValidateCompressedTexSubImage3DRobustANGLE(Context *context,
-                                                TextureType targetPacked,
+                                                TextureTarget targetPacked,
                                                 GLint level,
                                                 GLint xoffset,
                                                 GLint yoffset,
@@ -550,6 +550,18 @@
                                             GLsizei *length,
                                             GLuint64 *params);
 
+// GL_ANGLE_texture_external_update
+bool ValidateTexImage2DExternalANGLE(Context *context,
+                                     TextureTarget targetPacked,
+                                     GLint level,
+                                     GLint internalformat,
+                                     GLsizei width,
+                                     GLsizei height,
+                                     GLint border,
+                                     GLenum format,
+                                     GLenum type);
+bool ValidateInvalidateTextureANGLE(Context *context, TextureType targetPacked);
+
 // GL_ANGLE_texture_multisample
 bool ValidateTexStorage2DMultisampleANGLE(Context *context,
                                           TextureType targetPacked,
@@ -620,6 +632,11 @@
 bool ValidateMatrixLoadfCHROMIUM(Context *context, GLenum matrixMode, const GLfloat *matrix);
 bool ValidateMatrixLoadIdentityCHROMIUM(Context *context, GLenum matrixMode);
 
+// GL_CHROMIUM_lose_context
+bool ValidateLoseContextCHROMIUM(Context *context,
+                                 GraphicsResetStatus currentPacked,
+                                 GraphicsResetStatus otherPacked);
+
 // GL_CHROMIUM_path_rendering
 bool ValidateGenPathsCHROMIUM(Context *context, GLsizei range);
 bool ValidateDeletePathsCHROMIUM(Context *context, GLuint first, GLsizei range);
@@ -793,6 +810,69 @@
                                GLsizeiptr length,
                                GLbitfield access);
 
+// GL_EXT_memory_object
+bool ValidateBufferStorageMemEXT(Context *context,
+                                 TextureType targetPacked,
+                                 GLsizeiptr size,
+                                 GLuint memory,
+                                 GLuint64 offset);
+bool ValidateCreateMemoryObjectsEXT(Context *context, GLsizei n, GLuint *memoryObjects);
+bool ValidateDeleteMemoryObjectsEXT(Context *context, GLsizei n, const GLuint *memoryObjects);
+bool ValidateGetMemoryObjectParameterivEXT(Context *context,
+                                           GLuint memoryObject,
+                                           GLenum pname,
+                                           GLint *params);
+bool ValidateGetUnsignedBytevEXT(Context *context, GLenum pname, GLubyte *data);
+bool ValidateGetUnsignedBytei_vEXT(Context *context, GLenum target, GLuint index, GLubyte *data);
+bool ValidateIsMemoryObjectEXT(Context *context, GLuint memoryObject);
+bool ValidateMemoryObjectParameterivEXT(Context *context,
+                                        GLuint memoryObject,
+                                        GLenum pname,
+                                        const GLint *params);
+bool ValidateTexStorageMem2DEXT(Context *context,
+                                TextureType targetPacked,
+                                GLsizei levels,
+                                GLenum internalFormat,
+                                GLsizei width,
+                                GLsizei height,
+                                GLuint memory,
+                                GLuint64 offset);
+bool ValidateTexStorageMem2DMultisampleEXT(Context *context,
+                                           TextureType targetPacked,
+                                           GLsizei samples,
+                                           GLenum internalFormat,
+                                           GLsizei width,
+                                           GLsizei height,
+                                           GLboolean fixedSampleLocations,
+                                           GLuint memory,
+                                           GLuint64 offset);
+bool ValidateTexStorageMem3DEXT(Context *context,
+                                TextureType targetPacked,
+                                GLsizei levels,
+                                GLenum internalFormat,
+                                GLsizei width,
+                                GLsizei height,
+                                GLsizei depth,
+                                GLuint memory,
+                                GLuint64 offset);
+bool ValidateTexStorageMem3DMultisampleEXT(Context *context,
+                                           TextureType targetPacked,
+                                           GLsizei samples,
+                                           GLenum internalFormat,
+                                           GLsizei width,
+                                           GLsizei height,
+                                           GLsizei depth,
+                                           GLboolean fixedSampleLocations,
+                                           GLuint memory,
+                                           GLuint64 offset);
+
+// GL_EXT_memory_object_fd
+bool ValidateImportMemoryFdEXT(Context *context,
+                               GLuint memory,
+                               GLuint64 size,
+                               HandleType handleTypePacked,
+                               GLint fd);
+
 // GL_EXT_occlusion_query_boolean
 
 // GL_EXT_robustness
@@ -817,6 +897,39 @@
                             GLsizei bufSize,
                             void *data);
 
+// GL_EXT_semaphore
+bool ValidateDeleteSemaphoresEXT(Context *context, GLsizei n, const GLuint *semaphores);
+bool ValidateGenSemaphoresEXT(Context *context, GLsizei n, GLuint *semaphores);
+bool ValidateGetSemaphoreParameterui64vEXT(Context *context,
+                                           GLuint semaphore,
+                                           GLenum pname,
+                                           GLuint64 *params);
+bool ValidateIsSemaphoreEXT(Context *context, GLuint semaphore);
+bool ValidateSemaphoreParameterui64vEXT(Context *context,
+                                        GLuint semaphore,
+                                        GLenum pname,
+                                        const GLuint64 *params);
+bool ValidateSignalSemaphoreEXT(Context *context,
+                                GLuint semaphore,
+                                GLuint numBufferBarriers,
+                                const GLuint *buffers,
+                                GLuint numTextureBarriers,
+                                const GLuint *textures,
+                                const GLenum *dstLayouts);
+bool ValidateWaitSemaphoreEXT(Context *context,
+                              GLuint semaphore,
+                              GLuint numBufferBarriers,
+                              const GLuint *buffers,
+                              GLuint numTextureBarriers,
+                              const GLuint *textures,
+                              const GLenum *srcLayouts);
+
+// GL_EXT_semaphore_fd
+bool ValidateImportSemaphoreFdEXT(Context *context,
+                                  GLuint semaphore,
+                                  HandleType handleTypePacked,
+                                  GLint fd);
+
 // GL_EXT_texture_storage
 bool ValidateTexStorage1DEXT(Context *context,
                              GLenum target,
@@ -1018,6 +1131,70 @@
 // GL_OES_query_matrix
 bool ValidateQueryMatrixxOES(Context *context, GLfixed *mantissa, GLint *exponent);
 
+// GL_OES_texture_3D
+bool ValidateCompressedTexImage3DOES(Context *context,
+                                     TextureTarget targetPacked,
+                                     GLint level,
+                                     GLenum internalformat,
+                                     GLsizei width,
+                                     GLsizei height,
+                                     GLsizei depth,
+                                     GLint border,
+                                     GLsizei imageSize,
+                                     const void *data);
+bool ValidateCompressedTexSubImage3DOES(Context *context,
+                                        TextureTarget targetPacked,
+                                        GLint level,
+                                        GLint xoffset,
+                                        GLint yoffset,
+                                        GLint zoffset,
+                                        GLsizei width,
+                                        GLsizei height,
+                                        GLsizei depth,
+                                        GLenum format,
+                                        GLsizei imageSize,
+                                        const void *data);
+bool ValidateCopyTexSubImage3DOES(Context *context,
+                                  TextureTarget targetPacked,
+                                  GLint level,
+                                  GLint xoffset,
+                                  GLint yoffset,
+                                  GLint zoffset,
+                                  GLint x,
+                                  GLint y,
+                                  GLsizei width,
+                                  GLsizei height);
+bool ValidateFramebufferTexture3DOES(Context *context,
+                                     GLenum target,
+                                     GLenum attachment,
+                                     TextureTarget textargetPacked,
+                                     GLuint texture,
+                                     GLint level,
+                                     GLint zoffset);
+bool ValidateTexImage3DOES(Context *context,
+                           TextureTarget targetPacked,
+                           GLint level,
+                           GLenum internalformat,
+                           GLsizei width,
+                           GLsizei height,
+                           GLsizei depth,
+                           GLint border,
+                           GLenum format,
+                           GLenum type,
+                           const void *pixels);
+bool ValidateTexSubImage3DOES(Context *context,
+                              TextureTarget targetPacked,
+                              GLint level,
+                              GLint xoffset,
+                              GLint yoffset,
+                              GLint zoffset,
+                              GLsizei width,
+                              GLsizei height,
+                              GLsizei depth,
+                              GLenum format,
+                              GLenum type,
+                              const void *pixels);
+
 // GL_OES_texture_border_clamp
 bool ValidateGetSamplerParameterIivOES(Context *context,
                                        GLuint sampler,